### ISSUE TITLE
Add the possibility to modify a LedgerEntry linked to a Transition

### ISSUE THREAD

**Issue by matteo88** [befair/gasistafelice#177]
As for now it is not possible to modify a LedgerEntry which is not the last created one.

SimpleAccounting should be modified such that it will be possible to do so. This behavior also requires to modify the LedgerEntry.balance_current value of the modified LedgerEntry as well as all the following ones.

**Comment by feroda** [comment]
can you please provide some real world examples for the record?

**Comment by matteo88** [comment]
The following code keeps a LedgerEntry from being updated if it is not the last for the account:

```
    entries = self.account.ledger_entries.order_by('-entry_id')
    if self.pk:
        # WARNING: if self is not the last entry for the account
        # raise InvalidAccountingOperation
        if not len(entries) or self.pk != entries[0].pk:
            raise InvalidAccountingOperation(
                ugettext("You cannot modify an entry which is not the latter for this account" )
           )
```

So doing something like 
```
simple_accounting.utils.update_transaction(transaction, amount=updated_amount, date=updated_date)
```
will raise a LedgerEntry.InvalidAccountingOperation error:
```
In [5]: simple_accounting.utils.update_transaction(tx, amount=12)
---------------------------------------------------------------------------
InvalidAccountingOperation                Traceback (most recent call last)
/home/letti/.virtualenvs/gasdev2/local/lib/python2.7/site-packages/django/core/management/commands/shell.pyc in <module>()
----> 1 simple_accounting.utils.update_transaction(tx, amount=12)

/home/letti/.virtualenvs/gasdev2/src/django-simple-accounting/simple_accounting/utils.pyc in update_transaction(transaction, **kwargs)
    414             transaction.amount, kwargs['amount']
    415         )
--> 416         transaction.amount = kwargs['amount']
    417 
    418     if kwargs.has_key('date'):

/home/letti/.virtualenvs/gasdev2/src/django-simple-accounting/simple_accounting/models.pyc in _set_amount(self, amount)
   1036                 else:
   1037                     le.amount = amount
-> 1038                 le.save()
   1039 
   1040             if self.source.amount < 0:

/home/letti/.virtualenvs/gasdev2/src/django-simple-accounting/simple_accounting/models.pyc in save(self, *args, **kwargs)
   1166             if not len(entries) or self.pk != entries[0].pk:
   1167                 raise InvalidAccountingOperation(
-> 1168                     ugettext("You cannot modify an entry which is not the latter for this account"
   1169                 ))
   1170             else:

InvalidAccountingOperation: You cannot modify an entry which is not the latter for this account
```

**Comment by feroda** [comment]
This is exactly the real world example that I mean, I'd like to know in which use-case a cash account transaction done in the past has to be updated. And why it is not feasible to make a new transaction with the amount difference, given that the user could have already checked the cash account status and so have seen the transaction with the original (maybe wrong) amount.

This has to be know not only to decide if it worths to implement such a feature, but also to specify constraints for the implementation.

**Comment by domthu** [comment]
The major use case is for curtail families orders. It's a weekly use case.

**Comment by kobe25** [comment]
@domthu with "families" do you refer to Gasista Felice's "gasmembers"?

**Comment by matteo88** [comment]
@kobe25 yes, a real word "family" is usually referred as a single "gasmember" into the software.

@domthu says: "Originally, during the analysis, it was decided to allow updating of an existing transaction. That because there was the possibility that a referrer could modify a transaction several times during partial curtails, without creating new ones. We have made a lot of work to implement transaction references in order to update an existing transaction." 

Now we have to extend the economic module to introduce the new featured balance_current without compromising the original concept of transaction.

As a real world example, if we are modifying a transaction which is not the last one we should recompute the balance_current for all the newest transactions. This is possible by changing the behavior of the module.

**Comment by matteo88** [comment]
To make things clearer, I attach an image which shows how adding an  "update" feature of the economic module would change the present behavior of the software:

![update transaction](https://cloud.githubusercontent.com/assets/1458241/10845230/66ec6094-7f06-11e5-8f63-ca5291825d35.png)

The upper part of the image shows the current behavior, while its lower part shows the change in the  behavior after the modification.
